File: poo/Cpp/Esquerda/Esquerda.cpp
#include <iostream>
#include <string>

int main() {
    int n;
    std::cin >> n;
    std::string comandos;
    std::cin >> comandos;

    int direcao = 0;
    char direcoes[] = {'N', 'L', 'S', 'O'};

    for (char comando : comandos) {
        if (comando == 'D') {
            direcao = (direcao + 1) % 4;
        } else if (comando == 'E') {
            direcao = (direcao - 1 + 4) % 4;
        }
    }

    std::cout << direcoes[direcao] << std::endl;

    return 0;
}

File: poo/Java/Esquerda/Esquerda.java
import java.util.Scanner;

public class Esquerda {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String comandos = scanner.next();
        char[] direcoes = {'N', 'L', 'S', 'O'};
        int direcaoAtual = 0; 
        for (char comando : comandos.toCharArray()) {
            if (comando == 'D') {
                direcaoAtual = (direcaoAtual + 1) % 4;
            } else if (comando == 'E') {
                direcaoAtual = (direcaoAtual + 3) % 4;
            }
        }
        System.out.println(direcoes[direcaoAtual]);
        scanner.close();
    }
}